### PR TITLE
Feature ledger providers

### PR DESCRIPTION
**Pull request by bradleySuira**
This PR includes:
- a fix for what's new modal 
- getLedgerAccounts updates to work with the latest in the wallet
- add a migration to remove existing ledger accounts to clean the sate and start with the new accounts created
- add chainCode and publicKey to the client getter and factory to use it in bitcoin ledger provider
- remove all related to ledger bridge that we use

### CODE REVIEW

File: src/factory/client.ts [hunk 1]
```diff
@@ -436,21 +438,34 @@ export const createClient = (
   network: Network,
   mnemonic: string,
   accountType: AccountType,
-  derivationPath: string
+  derivationPath: string,
+  publicKey?: string,
+  chainCode?: string
 ) => {
   const assetData = cryptoassets[asset];
```
Comment (monokh) [inline src/factory/client.ts:444]: Can you do a check here that throws an error when publickey, chaincode is provided but client is not BTC?

File: src/factory/client.ts [hunk 2]
```diff
@@ -436,21 +438,34 @@ export const createClient = (
   network: Network,
   mnemonic: string,
   accountType: AccountType,
-  derivationPath: string
+  derivationPath: string,
+  publicKey?: string,
```
Comment (monokh) [inline src/factory/client.ts]: better put these 2 params into one object:

```
derivationPath: string,
bitcoinLedgerCache?: { publicKey: string, chainCode: string }
```

This makes the purpose of these options much clearer

File: src/store/state.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ const initialState: RootState = {
     notAskAgain: false,
   },
   experiments: {},
-  watsNewModalVersion: '',
+  whatsNewModalVersion: '',
```
Comment (monokh) [inline src/store/state.ts:50]: You have changed this but there is no migration, This means there will be an old state in the wallet called `watsNewModalVersion` 
Will this cause any bugs? I am happy without migration if it does not introduce errors when wallet updates.

File: src/store/getters.ts [hunk 4]
```diff
@@ -72,6 +74,8 @@ export default {
       useCache?: boolean;
       accountType?: AccountType;
       accountIndex?: number;
+      chainCode?: string | undefined;
```
Comment (monokh) [inline src/store/getters.ts]: `chainCode?: string | undefined`
is the same as `chainCode?: string` ? 
`?` includes undefined

Also make sure this parameter matches the same format as in `client.ts`

File: src/store/migrations/19_remove_existing_ledger_accounts.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Account, AccountType } from "../types";
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export const removeExistingLedgerAccounts = {
```
Comment (monokh) [inline src/store/migrations/19_remove_existing_ledger_accounts.ts:4]: Can you add a test for the migration :)
You can copy a wallet state from pre this version and test it

**Review by monokh — CHANGES_REQUESTED** [review]

File: src/factory/client.ts (hunk 2 above)
Comment (bradleySuira) [reply src/factory/client.ts]: done

File: src/store/getters.ts (hunk 4 above)
Comment (bradleySuira) [reply src/store/getters.ts]: fixed

File: src/store/state.ts (hunk 3 above)
Comment (bradleySuira) [reply src/store/state.ts:50]: added inside the last migration now

File: src/factory/client.ts (hunk 1 above)
Comment (bradleySuira) [reply src/factory/client.ts:444]: it works, tested with the wallet and evm accounts

File: src/store/migrations/19_remove_existing_ledger_accounts.ts (hunk 5 above)
Comment (bradleySuira) [reply src/store/migrations/19_remove_existing_ledger_accounts.ts:4]: @monokh added this way to test the state, also you can run it `npx ts-node ./src/store/migrations/19_remove_existing_ledger_accounts_test.ts` but to be honest I think that can be better if I try later to find a way to test all migrations and don't need to upload any json files or maybe we can set a folder for all states and examples per version.

Screen Shot 2022-05-25 at 7 27 45 AM

File: src/factory/client.ts (hunk 1 above)
Comment (monokh) [reply src/factory/client.ts:444]: I'm sure it works, but my suggetion gives some feedback to a consumer that would call those parameters on a non BTC client, and expect it to work.

File: src/store/state.ts (hunk 3 above)
Comment (monokh) [reply src/store/state.ts:50]: I dont see this variable being treated in the migration 😄  did you commit?

**Review by monokh — APPROVED** [review]

**Pull request merged**